### PR TITLE
Handyman Profile | Page

### PR DESCRIPTION
**Pull request by hashemt**
create profile page
edit the style of stars in profile component
Relates #14

### CODE REVIEW

File: client/src/components/Pages/Profile/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+
+import ProfileCard from '../../common/ProfileCard';
+
+class Profile extends React.Component {
+  state = {
+    profileData: {},
+    statusCode: '',
```
Comment (AbdallahAmmar96) [inline client/src/components/Pages/Profile/index.js]: delete the status code

File: client/src/components/Pages/Profile/index.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  };
+
+  async componentDidMount() {
+    try {
+      const { id } = this.props;
+      const response = await axios.get(`/api/v1/profile/${id}`);
+      this.setState({
+        profileData: response.data.data,
+        statusCode: response.data.statusCode,
```
Comment (AbdallahAmmar96) [inline client/src/components/Pages/Profile/index.js]: no need to put the status code in the state

File: client/src/components/common/ProfileCard/style.css [hunk 3]
```diff
@@ -28,6 +28,13 @@
     background-color: linear;
 }
 
+.stars{
+    width: 40%;
+    display: flex;
+    justify-content: center;
+
```
Comment (AbdallahAmmar96) [inline client/src/components/common/ProfileCard/style.css]: delete this line

File: client/src/components/Pages/Profile/index.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  async componentDidMount() {
+    try {
+      const { id } = this.props;
+      const response = await axios.get(`/api/v1/profile/${id}`);
+      if (response.data.data) {
+        this.setState({
+          profileData: response.data.data,
+        });
+      } else {
+        const { history } = this.props;
```
Comment (AbdallahAmmar96) [inline client/src/components/Pages/Profile/index.js]: Show the user a notification and then redirect to any page

File: client/src/components/Pages/Profile/index.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        history.push('/404');
+      }
+    } catch (e) {
+      const { history } = this.props;
```
Comment (AbdallahAmmar96) [inline client/src/components/Pages/Profile/index.js]: Make the destruction  one time at the top

Comment (AbdallahAmmar96) [inline client/src/components/Pages/Profile/index.js]: And send a notification for the user

File: server/controllers/routes/getProfile.js [hunk 6]
```diff
@@ -3,7 +3,8 @@ const { getProfile } = require('../../db/queries/getProfile');
 module.exports = async (req, res, next) => {
   try {
     const profileData = await getProfile(req.params.id);
-    res.send({ data: profileData.rows[0], statusCode: 200 });
+    if (profileData.rows[0]) res.send({ data: profileData.rows[0], statusCode: 200 });
+    else res.send({ message: 'user not found', statusCode: 200 });
```
Comment (mossa-Sammer) [inline server/controllers/routes/getProfile.js]:
```suggestion
    else res.send({ message: 'user not found', statusCode: 400 });
```

File: client/src/components/common/ProfileCard/index.js [hunk 7]
```diff
@@ -36,9 +36,11 @@ const ProfileCard = ({
         </div>
       </div>
       <div className="card-btnContainer">
-        {starNumber.map(() => (
-          <Star />
-        ))}
+        <div className="stars">
+          {starNumber.map((item, index) => (
+            <Star key={index} />
```
Comment (mossa-Sammer) [inline client/src/components/common/ProfileCard/index.js]: try to avoid using map index

File: client/src/components/Pages/Profile/index.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import { NotificationManager } from 'react-notifications';
```
Comment (mossa-Sammer) [inline client/src/components/Pages/Profile/index.js]:
```suggestion
import { NotificationManager as notify } from 'react-notifications';
```

File: client/src/components/common/ProfileCard/index.js (hunk 7 above)
Comment (AbdallahAmmar96) [reply client/src/components/common/ProfileCard/index.js]: he needs to use it here

**Pull request merged**